### PR TITLE
Phenotype

### PR DESCRIPTION
**Pull request by sierra-moxon**

### CODE REVIEW

**Comment by sierra-moxon** [comment]
This PR adds the phenotype loader and the refactor of dataProvider -- however, I've commented out the call of the phenotype loader in aggregate_loader.py for now, in order to do some more self-testing of the time increases.  I think it will be ok, but I also want the data refresh to go thru the entire pipeline before Friday code freeze.  I also pushed through a dataProvider with just one value in addition to the dataProvider list, so that gene descriptions will be ok for 1.6.  We'll refactor this (and the allele table on the gene page) in 1.7.  Any chance you have a few minutes to review this so we can test it in the build process tomorrow @christabone ?  I'll make a new PR request for turning on the phenotype load (hopefully before friday).

**Comment by sierra-moxon** [comment]
For reference, pheno load adds about an hour to the load (locally, likely faster on the build server -- I usually see about a 25% speed up vs. locally -- but its hard for me to say for certain how long it will really take on a faster machine).  MGI especially has a lot of pheno annotations.  We should give it a try on the build server.  But, this PR just updates the existing data and schema according to the dataProvider changes and such.  @christabone.  It would be good to get it into build and then enable phenotype loader next.

File: src/loaders/transactions/disease_gene.py [hunk 1]
```diff
@@ -10,68 +10,66 @@ def disease_gene_object_tx(self, data):
...
                 SET dga :DiseaseEntityJoin
-
+                SET dga.dataProviders = row.dataProviders
+                SET dga.dataProvider = row.dataProvider
+                
+            MERGE (dfa)-[dfal:LOADED_FROM]-(l)
+                
             FOREACH (rel IN CASE when row.relationshipType = 'is_marker_for' THEN [1] ELSE [] END |

                 MERGE (gene)<-[fafg:IS_MARKER_FOR {uuid:row.uuid}]->(d)
```
Comment (christabone) [inline src/loaders/transactions/disease_gene.py]: Any idea about these weird characters? I see them on a few lines below.

File: src/extractors/bgi_ext.py [hunk 2]
```diff
@@ -2,19 +2,45 @@
...
+            for dataProviderPage in dataProviderPages:
+                crossRefCompleteUrl = UrlService.get_page_complete_url(dataProvider, xrefUrlMap, dataProvider,
+                                                                       dataProviderPage)
+                dataProviderCrossRefSet.append(
+                    CreateCrossReference.get_xref(dataProvider, dataProvider, dataProviderPage,
+                                                  dataProviderPage, dataProvider, crossRefCompleteUrl,
+                                                  dataProvider + dataProviderPage))
+
+            dataProviders.append(dataProvider)
+            print ("data provider: " + dataProvider)
```
Comment (christabone) [inline src/extractors/bgi_ext.py]: Are we keeping these print statements in the loader?

**Comment by christabone** [comment]
Ok, totally fine with trusting Sierra on this one. It's a big merge. We should plan to remove those funky characters in the near future.

File: src/loaders/transactions/disease_gene.py (hunk 1 above)
Comment (sierra-moxon) [reply src/loaders/transactions/disease_gene.py]: @christabone - sorry which weird characters?  I'm not seeing them...

Comment (christabone) [reply src/loaders/transactions/disease_gene.py]: They're a bit strange -- I took a screenshot:

image

**Pull request merged**